cmd/state-installer/cmd.go

```diff
@@ -439,7 +449,7 @@ func determineLegacyUpdate(stateToolInstalled bool, packagedStateExe, payloadPat
 	// Detect whether this is a fresh install or an update
 	var isUpdate bool
 	switch {
-	case (params.sourceInstaller == "install.sh" || params.sourceInstaller == "install.ps1") && fileutils.FileExists(packagedStateExe):
+	case (params.sourceInstaller == "install.sh" || params.sourceInstaller == "install.ps1" || noArgs()) && fileutils.FileExists(packagedStateExe):
```

Comment:
This is gonna conflict with your PR about handling updates with the update flag.

Reply:
That PR is already in master and this branch is update to date with it.

Reply:
ooh this is in the legacy code. I don't think double clicking would ever reach this considering it'd only get here if you pass legacy flags?

Reply:
It will still hit this as the update flag isn't set. Adding the `noArgs` call here just ensures we don't get the update UX.